src/test/java/io/vertx/core/http/HttpTest.java

```diff
@@ -2704,12 +2705,13 @@ public void testGetAbsoluteURI() {
 
   @Test
   public void testListenInvalidPort() throws Exception {
-    /* Port 7 is free for use by any application in Windows, so this test fails. */
-    Assume.assumeFalse(System.getProperty("os.name").startsWith("Windows"));
     server.close();
-    server = vertx.createHttpServer(new HttpServerOptions().setPort(7));
-    server.requestHandler(noOpHandler()).listen(onFailure(server -> testComplete()));
-    await();
+    try(ServerSocket occupied = new ServerSocket(0)){
+      occupied.setReuseAddress(false);
+      server = vertx.createHttpServer(new HttpServerOptions().setPort(occupied.getLocalPort()));
+      server.requestHandler(noOpHandler()).listen(onFailure(server -> testComplete()));
+      await();
+    }
```

Comment:
the occupied server socket should be closed in a finally block surrounding this

Reply:
Please correct me if I am wrong but:

- The `occupied` resource is opened in the [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)
- ServerSocket implements java.io.Closeable which extends AutoCloseable
- The `await()` will prevent the try-with-resources AutoClosing the socket before
the test has executed. 

I do see the code is not obvious if read quickly and I tried to talk to this above in ["The socket is Closable so the try with resources should close the port (after the await) "](https://github.com/eclipse-vertx/vert.x/pull/3443#issue-433582297). I apologise that I did not put a comment this effect in the code (and also for the use of port `0` result of assigning a free port) but the rest of the tests are not very wordy so I was trying to adopt the same style.

Reply:
ServerSocket is autoclosable and occupied is closed by the try-with-resources statement

Reply:
indeed that's very correct, I am not using this pattern as we don't use it often with non blocking resources